src/infinite-scroll.js

```diff
@@ -131,7 +129,7 @@ angular.module(MODULE_NAME, [])
         return throttled;
       }
 
-      const handler = (THROTTLE_MILLISECONDS != null) ?
```

Comment:
sure, what should actually happen is something like:

```
async function doScroll() {
    ...
    while(shouldScroll() && scrollEnabled) {
        result = scope.infiniteScroll();
        if (typeof result.then === 'function') {
            await scope.infiniteScroll();
            await $timeout(THROTTLE_MILLISECONDS);
        } else {
            await $timeout(THROTTLE_MILLISECONDS);
            return;
        }
    }
    ...
}
function getHandler() {
     let prom = null;
     async function resetProm(doScrollProm) {
         await doScrollProm;
         prom = null;
     }
     async function innerHandler() {
        if (!prom) {
             prom = resetProm(doScroll());
        }
        await prom;
    }
    return innerHandler;
}

const handler = getHandler();
```

The idea, throttling is all handled by doScroll, every time it's called it loops until the container is full, while handling the throttling. However if infiniteScroll is a regular funciton with no promise, doScroll just defaults to the previous behaviour.

How to convert this from ES2017 async/await into $q then chains and support all the other features and checks I've left out is left as an exercise for the reader.